app/controllers/application_controller.rb

```diff
@@ -367,6 +367,7 @@ def report_data
     if options.nil? || options[:view].nil?
       model_view = process_params_model_view(params, options)
       @edit = session[:edit]
+      options_from_session(options)
```

Comment:
IMHO, while we are all striving for faster page loads, executing the same set of queries, twice, back-to-back to load the listviews just does *not* seem right.
Hope we can address this quickly.